### ISSUE TITLE
eventlistener processes are never started

### ISSUE THREAD

**Issue by rwlincoln** [ochinchina/supervisord#71]
In the `getAllProcess` method of `ProcessManager` only the `procs` field is iterated over:

https://github.com/ochinchina/supervisord/blob/354e8ceb112f19a7a28e8b84252edb25859b6ae7/process/process_manager.go#L129-L131

Processes created from `eventlistener` sections are added to the `eventListeners` field. As a result, the event listener processes are not started in `StartAutoStartPrograms`...

https://github.com/ochinchina/supervisord/blob/354e8ceb112f19a7a28e8b84252edb25859b6ae7/process/process_manager.go#L36-L40

...and the `registerEventListener` method is never called:

https://github.com/ochinchina/supervisord/blob/354e8ceb112f19a7a28e8b84252edb25859b6ae7/process/process.go#L540-L548

Perhaps the solution is to iterate over both process maps in the manager:

```go
func (pm *ProcessManager) getAllProcess() []*Process {
	tmpProcs := make([]*Process, 0)
	for _, proc := range pm.procs {
		tmpProcs = append(tmpProcs, proc)
	}
	for _, eventListener := range pm.eventListeners {
		tmpProcs = append(tmpProcs, eventListener)
	}
	return sortProcess(tmpProcs)
}
```

**Comment by ochinchina** [comment]
Thank you for finding this issue and I have fixed this issue.
The solution is a little different from yours. The getAllProcess() should only returns the program (not include the event listeners) because some operations can be applied to program only.

**Comment by rwlincoln** [comment]
Fixed in 37d8b9736f27433a5d24ec32345626b02d45e834.
